esx_service/vmdk_ops.py

```diff
@@ -153,12 +157,24 @@ def RunCommand(cmd):
 
     return (s, o)
 
+def is_valid_volume_name(name):
+    """ Check the given name is a valid volume name """
+    pattern = "^" + VALID_VOLUME_NAME_REG + "$"
+    valid_name_reg = re.compile(pattern)
+    if valid_name_reg.match(name):
+        return True
+    else:
+        return False
 
 # returns error, or None for OK
 # opts is  dictionary of {option: value}.
 # for now we care about size and (maybe) policy
 def createVMDK(vmdk_path, vm_name, vol_name, opts={}, vm_uuid=None, tenant_uuid=None, datastore=None):
     logging.info("*** createVMDK: %s opts = %s", vmdk_path, opts)
+    if not is_valid_volume_name(vol_name):
```

Comment:
We have a util function parse_vol_name, which check whether the vol_name is match r"^.*-[0-9]{6}$" . So there is no limit on which character will be included in the vol_name. But docker local driver limit vol_name can only be "[a-zA-Z0-9][a-zA-Z0-9_.-]" . Do we want have the same limit as docker local driver does?